### ISSUE TITLE
Minecraft RTX?

### ISSUE THREAD

**Issue by soshimee** [ChristopherHX/mcpelauncher-manifest#248]
Is there a way to play Minecraft RTX on Linux?
If there is, how do I do it?

**Comment by chinoto** [comment]
If there is, I doubt it's by using an Android client. Some of the code for RTX may have slipped into non-Windows 10 clients as dead code because they share a code base, but the implementation is DirectX based, while the Android version renders via OpenGL ES. Your best bet is a Windows 10 VM if the GPU passthrough works properly (I don't do much with VMs). Otherwise, implement the UWP (Universal Windows Program) API in WINE.

**Comment by soshimee** [comment]
Would implementing UWP help? Since I have to change the version in some hub it might glitch out.

**Comment by soshimee** [comment]
![image](https://user-images.githubusercontent.com/45235204/120886723-56ac7180-c62a-11eb-94c6-b843006842d1.png)
Hmm, maybe I can FORCE this.

**Comment by chinoto** [comment]
> Would implementing UWP help? Since I have to change the version in some hub it might glitch out.

Probably, but that is quite a lot of work. [Wine](https://www.winehq.org/) has existed for years and still hasn't fully implemented the Win32 API, though they have implemented what is needed for most programs that people care about. Just to be clear, when I say "implement", I mean you have to actually write code, which could take months even if you have prior experience using the API.

> Hmm, maybe I can FORCE this.

No, like I said the Android version of Minecraft renders using OpenGL and the ray tracing is only implemented in DirectX. Before you ask, no, you [can't jam DirectX code into OpenGL](https://stackoverflow.com/questions/7686619/mixing-directx-and-opengl) and expect it to work.

**Comment by soshimee** [comment]
Can I then make it work in DirectX?

**Comment by chinoto** [comment]
No, the Android version is rendering with OpenGL. The only ways to make this work that I can think of is to put in the work to make Wine support UWP apps, somehow get Mojang to directly support Linux, or run in a Windows VM.
